components/powerMgr/linux/le_ulpm.c

```diff
@@ -497,15 +497,20 @@ le_result_t le_ulpm_ShutDown
     if (pm_CheckWakeLock())
     {
         LE_ERROR("Wakelock held!! System can't be shut down. Try again.");
-        return LE_UNAVAILABLE;
+        return LE_NOT_POSSIBLE;
     }
 
     le_framework_NotifyExpectedReboot();
 
+    // Convert shutdown value to a string
+    char shutdown_value_str[12];
+    if (sprintf(shutdown_value_str, "%d", shutdown_value) < 0)
```

Comment:
Could you use something like snprintf instead? Although unlikely here, it is still best practice to handle buffer overflow case.